main.tf

```diff
@@ -0,0 +1,191 @@
+  name = "${local.instance_alias}"
+}
+
+resource "aws_iam_role" "shuntingyard_task_exec" {
+  name = "${local.instance_alias}-ecs-task-exec-${var.aws_region}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "task_exec_managed" {
+  role       = "${aws_iam_role.shuntingyard_task_exec.id}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "shuntingyard_task" {
+  name = "${local.instance_alias}-ecs-task-${var.aws_region}"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "ecs-tasks.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "s3_for_shuntingyard" {
+  name  = "s3"
+  role  = "${aws_iam_role.shuntingyard_task.id}"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:DeleteObject",
+                "s3:DeleteObjectVersion",
+                "s3:Get*",
+                "s3:List*",
+                "s3:PutBucketLogging",
+                "s3:PutBucketNotification",
+                "s3:PutBucketVersioning",
+                "s3:PutObject",
+                "s3:PutObjectAcl",
+                "s3:PutObjectTagging",
+                "s3:PutObjectVersionAcl",
+                "s3:PutObjectVersionTagging"
+            ],
+            "Resource": [
+                          "${join("\",\"", formatlist("arn:aws:s3:::%s",var.allowed_s3_buckets))}",
+                          "${join("\",\"", formatlist("arn:aws:s3:::%s/*",var.allowed_s3_buckets))}"
+                        ]
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy" "sqs_for_shuntingyard" {
+  name  = "sqs"
+  role  = "${aws_iam_role.shuntingyard_task.id}"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": {
+        "Effect": "Allow",
+        "Action": [
+          "sqs:SendMessage",
+          "sqs:ReceiveMessage",
+          "sqs:DeleteMessage"
+        ],
+        "Resource": "${aws_sqs_queue.shuntingyard_sqs_queue.arn}"
+    }
+}
+EOF
+}
+
+resource "aws_cloudwatch_log_group" "shuntingyard_ecs" {
+  name = "${local.instance_alias}"
+  tags = "${var.tags}"
+}
+
+data "template_file" "shuntingyard_config_yaml" {
+  template = "${file("${path.module}/templates/shunting-yard-config.yml.tmpl")}"
+
+  vars {
+    source_metastore_uri      = "${var.source_metastore_uri}"
+    target_metastore_uri      = "${var.target_metastore_uri}"
+    shunting_yard_sqs_queue   = "${aws_sqs_queue.shuntingyard_sqs_queue.id}"
```

Comment:
`shunting_yard` --> `shuntingyard` for consistency?

Reply:
done.